src/Razor/Microsoft.NET.Sdk.Razor/src/build/netstandard2.0/Microsoft.NET.Sdk.Razor.Components.Wasm.targets

```diff
@@ -305,18 +300,22 @@ Copyright (c) .NET Foundation. All rights reserved.
     </Copy>
   </Target>
 
-  <Target Name="_CreateBlazorTrimmerRootDescriptorFiles" BeforeTargets="ILLink" AfterTargets="ComputeResolvedFilesToPublishList">
+  <Target Name="_BlazorWasmPrepareForLink" BeforeTargets="PrepareForILLink">
     <PropertyGroup>
       <_BlazorTypeGranularTrimmerDescriptorFile>$(IntermediateOutputPath)typegranularity.trimmerdescriptor.xml</_BlazorTypeGranularTrimmerDescriptorFile>
     </PropertyGroup>
 
     <ItemGroup>
       <_BlazorTypeGranularAssembly
-          Include="@(ResolvedFileToPublish)"
+          Include="@(ManagedAssemblyToLink)"
           Condition="'%(Extension)' == '.dll' AND ($([System.String]::Copy('%(Filename)').StartsWith('Microsoft.AspNetCore.')) or $([System.String]::Copy('%(Filename)').StartsWith('Microsoft.Extensions.')))">
         <Required>false</Required>
         <Preserve>all</Preserve>
       </_BlazorTypeGranularAssembly>
+
+      <ManagedAssemblyToLink
+        IsTrimmable="true"
+        Condition="'%(Extension)' == '.dll' AND ($([System.String]::Copy('%(Filename)').StartsWith('Microsoft.AspNetCore.')) or $([System.String]::Copy('%(Filename)').StartsWith('Microsoft.Extensions.')))" />
```

Comment:
(minor) I think there is a way to not duplicate this condition, and instead just use `@(_BlazorTypeGranularAssembly)` to update these `ManagedAssemblyToLink` items.

Reply:
Is that by removing and adding it? ItemGroup `Update` doesn't work inside of a target.